root/static/scripts/common/components/sidebar/AcousticBrainz.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import hydrate from '../../../../../utility/hydrate';
+import {SidebarProperty, SidebarProperties} from
+  '../../../../../layout/components/sidebar/SidebarProperties';
+
+const SidebarAcousticBrainz = ({recording}: {recording: RecordingT}) => {
+  const [count, setCount] = React.useState(0);
+  const [data, setData] = React.useState(null);
+
+  function fetchAcousticBrainzCount() {
+    const countUrl = `//acousticbrainz.org/api/v1/${recording.gid}/count`;
+    fetch(countUrl).then(
+      resp => resp.json(),
+    ).then(
+      data => {
+        setCount(data.count);
+        if (data.count) {
+          fetchAcousticBrainzData();
+        }
+      },
+    );
+  }
+
+  function fetchAcousticBrainzData() {
+    const dataUrl = `//acousticbrainz.org/api/v1/${recording.gid}/low-level`;
+    fetch(dataUrl).then(
+      resp => resp.json(),
+    ).then(
+      data => {
+        setData(data);
+      },
+    );
+  }
+
+  React.useEffect(fetchAcousticBrainzCount, [recording.gid]);
+  return count ? (
+    <>
+      <h2>
+        {l('Acoustic information')}
+      </h2>
+
+      <a className="external" href={`//acousticbrainz.org/${recording.gid}`}>
+        {texp.ln(
+          '{count} submission on AcousticBrainz',
+          '{count} submissions on AcousticBrainz',
+          count,
+          {count},
+        )}
+      </a>
+
+      {data === null ? null : (
+        <SidebarProperties>
+
+          {data.tonal.key_strength > 0.5 ? (
+            <SidebarProperty className="acousticbrainz_key" label={l('Key:')}>
+              {`${data.tonal.key_key} ${data.tonal.key_scale}`}
```

Comment:
just one thing to keep in mind here - we use the sharp variations of all keys, so Bb major will be reported as A# major. This means that some people might complain that it's "wrong". We could provide mappings here, so we always report "A#/Bb major" in these cases, for example.

I'm not sure what the best strength threshold is. I was reading [the paper that describes the algorithm](https://www.researchgate.net/publication/220669391_Tonal_Description_of_Polyphonic_Audio_for_Music_Content_Processing) but it doesn't show how the accuracy increases as the strength threshold is altered. I'll discuss this with my coworkers. For now 0.5 is probably a "safe" bet.

Do we want a way for people to complain? Or at the very least a popup that says "this is automatically generated and might be wrong"?